rholang/src/main/scala/coop/rchain/rholang/interpreter/Env.scala

```diff
@@ -0,0 +1,32 @@
+package coop.rchain.rholang.interpreter
+
+import monix.eval.Task
+
+import scala.collection.immutable.HashMap
+
+object Env {
```

Comment:
I still don't understand this construction.
How is this better than just creating an Env class?

Reply:
Ok. Say I decide to make a class for `Env`. I want that class to have all library functions for `immutable.HashMap` because this implementation is a hash map and they'd be very handy to have. If I try to do `class Env extends HashMap[Int, Par]`, I get an error: `Illegal inheritance from sealed trait HashMap`. I could create a wrapper class that implements the same methods with an internal hash map, but it's a bit sloppy. So, the alternative is to introduce `implicit class DeBruijn`, but I can't do a top level implicit class. I have to put it in a singleton. Think of `Env` as the place where all possible environment implementations live.

Reply:
You've explained why, but not what the code actually does. I understand that you want an implicit. I don't see how that gets you from A to B. Why can you add a parameter to Env? Etc. This looks like magic to me, and I won't approve it until I understand it. Also, you're going to need to add comments to the code explaining what is going on.